namerd/iface/mesh/src/test/scala/io/buoyant/namerd/iface/mesh/ResolverServiceTest.scala

```diff
@@ -0,0 +1,129 @@
+package io.buoyant.namerd.iface.mesh
+
+import com.twitter.finagle.stats.NullStatsReceiver
+import com.twitter.finagle._
+import com.twitter.io.Buf
+import com.twitter.util.{Activity, Var}
+import io.buoyant.test.FunSuite
+import io.linkerd.mesh.Endpoint.AddressFamily
+import io.linkerd.mesh.{Endpoint, Replicas, ReplicasReq, Path => MPath}
+
+class ResolverServiceTest extends FunSuite {
+
+  test("getReplicas") {
+    val pfx = Path.read("/#/atl")
+    val states = Var[Activity.State[NameTree[Name.Bound]]](Activity.Pending)
+    val namers = Map(pfx -> new Namer { def lookup(path: Path) = Activity(states) })
+    val resolver = new ResolverService.ServerImpl(namers, NullStatsReceiver)
+
+    val vaddr = Var[Addr](Addr.Bound(Address("1.2.3.4", 7777)))
+    states() = Activity.Ok(NameTree.Leaf(Name.Bound(vaddr, Path.read("/#/atl/slime/season"))))
+
+    val path = Path.read("/#/atl/slime/season")
+    val req = ReplicasReq(Some(MPath(path.elems)))
+
+    val endpoint = Endpoint(
+      inetAf = Some(AddressFamily.INET4),
+      address = Some(Buf.ByteArray.Owned(Array[Byte](1, 2, 3, 4))),
+      port = Some(7777),
+      meta = Some(Endpoint.Meta(nodeName = None))
+    )
+    val rsp = await(resolver.getReplicas(req))
+    assert(rsp.result.get == Replicas.OneofResult.Bound(Replicas.Bound(Seq(endpoint))))
+  }
+
+  test("streamReplicas") {
```

Comment:
TIOLI but it would be nice if these test names were more descriptive than just the name of the method being tested...